### DISCUSSION TITLE
Progress Bar

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#287] [Component suggestions]
<img width="480" alt="visuele weergave van de progress bar component" src="https://github.com/user-attachments/assets/716be4b9-adcd-4717-9d60-cc496b34e985">

We willen graag meer leren over de Progress Bar component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, GitHub of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack]({url-slack-bericht})

## Naam
Progress Bar

## Doel
Toont de voortgang van een proces in een balk.

## Ook bekend als
Progress, Meter, Linear Progress, Progress Indicator

## Gerelateerde componenten
[Spinner](https://github.com/orgs/nl-design-system/discussions/254)

## Onderzoek

[`<progress>`: The Progress Indicator element - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress)
Documentatie van Mozilla Developers Network over het `<progress>` HTML element.

[A study on tolerable waiting time: how long are Web users willing to wait? - ResearchGate](https://www.researchgate.net/publication/240624333_A_study_on_tolerable_waiting_time_how_long_are_Web_users_willing_to_wait_Citation_Nah_F_2004_A_study_on_tolerable_waiting_time_how_long_are_Web_users_willing_to_wait_Behaviour_Information_Technology_f)
Uit een onderzoek van Fiona Fui-Hoon Nah bleek dat gebruikers bereid zijn langer te wachten wanneer ze feedback krijgen, zoals een voortgangsbalk. De mediane wachttijd was 22,6 seconden voor gebruikers met een voortgangsbalk, terwijl dit slechts 9 seconden was voor degenen zonder zicht op de voortgang.

[Validatie onderzoek Loket Buitenland - PDF (3,6 MB)](https://github.com/azdpc/prototype-1overheid/blob/master/rapportage-loket-buitenland-ruigrok-netpanel-publicatie-v1.0.pdf)
Tijdens een gebruikersonderzoek van 1Overheid werd een Progress Bar in een meerstappenformulier gebruikt. In de rapportage is daar het volgende over te genoteerd:

> De voortgangsbalk maakt duidelijk hoe ver gebruikers in het proces zijn. En dat vinden gebruikers prettig.

**Comment by jeffreylauwers** [comment]
## Den Haag

Niet als 'los' component maar met de naam 'Form Progress Bar' als onderdeel van de component Form Progress (Storybook)

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-input-form-progress--docs)
- [Figma](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=16940-96956&t=aH29blg7MRO8SRaJ-4)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/FormProgress)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-12-23 om 09 17 09](https://github.com/user-attachments/assets/9a0acb3f-0fdc-4567-af7d-9188cf7e2739)

---

### Notities

Visueel is de Progress Bar 2px hoog met lichtgrijze achtergrond. De voortgang wordt aangegeven met groen.

**Comment by jeffreylauwers** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/67.2.0/components/on-page-interactives/progress-bar)
- [Storybook](https://storybook.dso-toolkit.nl/67.2.0/?path=/docs/html-css-progress-bar--docs)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/progress-bar)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-12-23 om 09 30 36](https://github.com/user-attachments/assets/d91e410d-8c1b-46db-ac02-1da75467b1f8)

Arbitrary Values
![Scherm­afbeelding 2024-12-23 om 09 30 26](https://github.com/user-attachments/assets/67d929e3-7bd9-4518-a5ae-966185873589)

---

### Notities

In de documentatie worden voortgangsindicatoren (Spinner / Progress Bar) in het algemeen beschreven. Daarbij wordt een mooie uiteenzetting gedaan wanneer je welk component zou moeten inzetten:

> Richtlijnen bij gebruik spinner/progress bar:
>
>- wachttijd kleiner dan 1 seconde: geen spinner
> - wachttijd tussen 1 en 4 seconden: spinner (voorbeeld boven)
> - wachttijd groter dan 4 seconden: progress bar (voorbeelden beneden)
> - wachttijd tussen 4 seconden en 1 minuut: progress bar met percentage
> - wachttijd groter dan 1 minuut: progress bar met tijd inschatting

Specifiek voor de Progress Bar er iets geschreven over `aria-valuenow`:

> Probeer altijd de progress waarde in aria-valuenow op te nemen, zodat de gebruiker weet hoe lang hij/zij moet wachten. Omdat het label ook de 'accessible name' is voor het component, dient daarin idealiter ook genoemd te worden 'waarop' de gebruiker aan het wachten is.

Visueel is de Progress Bar 16px hoog met groene border en witte achtergrond. De voortgang wordt aangegeven met groen.

**Comment by jeffreylauwers** [comment]
## DUO

Niet als 'los' component maar met de naam 'File Progress' als onderdeel van de component File Upload.

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/file-upload)

---

### Screenshots / Varianten

**Documentatie website**

![Scherm­afbeelding 2024-12-23 om 10 22 11](https://github.com/user-attachments/assets/ad382ed5-577b-41ad-a654-59ec226548e5)

---

### Notities

- 3 verschillende varianten: Default (blauw), Invalid (rood) en Completed (groen).
- Percentage staat vermeld binnen de balk zelf.
- Visueel is de Progress Bar 16px hoog met lichtgrijze achtergrond. De voortgang wordt aangegeven met blauw. Bij 'invalid' of 'completed' krijgt de voortgang een andere kleur.

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

[Patternlab](https://patternlab.lenteveld.nl/patterns/02-organisms-200-Miscellaneous-130-Project-Information-OpenGemeentenProjectInformation-Page/02-organisms-200-Miscellaneous-130-Project-Information-OpenGemeentenProjectInformation-Page.html)

---

### Screenshots / Varianten

**Patternlab**

Project Information Indicator
![Scherm­afbeelding 2024-12-23 om 09 47 01](https://github.com/user-attachments/assets/685095da-418f-42d6-9ee6-0c1b70fb3047)

Project Information Page
![Scherm­afbeelding 2024-12-23 om 09 48 32](https://github.com/user-attachments/assets/dcae61d1-d376-4414-9766-6a14c299092c)

---

### Notities

- Onder de naam: Project Information Indicator
- Als onderdeel van een contentblok dat informatie geeft over de voortgang van een project.
- Visueel is de Progress Bar 16px hoog met donkergrijze border en witte achtergrond. De voortgang wordt aangegeven met donkerblauw.

**Comment by jeffreylauwers** [comment]
## Nijmegen

Niet als 'los' component maar met de naam 'Progress Bar' als onderdeel van een meerstappenformulier.

[Fix je shit - nijmegen.nl](https://www.nijmegen.nl/fixjeshit)

---

### Screenshots / Varianten

**nijmegen.nl**

![Scherm­afbeelding 2024-12-23 om 10 36 28](https://github.com/user-attachments/assets/800ada94-3c89-40b7-b3e5-f86cff4a1345)

---

### Notities

Visueel is de Progress Bar 16px hoog met lichtgrijze achtergrond. De voortgang wordt aangegeven met donkergroen. De balk heeft afgeronde hoeken.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Progress Bar op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=Progress+Bar)
